src/silx/gui/dialog/_ColormapPercentileWidget.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+from silx.gui import qt
+from silx.gui.utils import blockSignals
+
+
+class ColormapPercentilesWidget(qt.QWidget):
+    """
+    Widget to define the percentiles to be used when computing the colormap in autoscale / percentile mode.
+
+    A scalar value (that can be seen as saturation) is defined by the user and then converted to percentiles using the 'fromSaturationToPercentiles' function.
+    """
+
+    percentilesChanged = qt.Signal(tuple)
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.setLayout(qt.QHBoxLayout())
+
+        self._slider = qt.QSlider(qt.Qt.Horizontal, self)
+        self.layout().addWidget(self._slider)
+
+        self._spinBox = qt.QSpinBox(self)
+        self.layout().addWidget(self._spinBox)
+
+        self._setRange(0, 100)
+
+        # connect signal / slot
+        self._slider.valueChanged.connect(self.setSaturationValue)
+        self._spinBox.valueChanged.connect(self.setSaturationValue)
+
+    def setSaturationValue(self, value: int):
+        with blockSignals(self._slider, self._spinBox):
+            self._slider.setValue(value)
+            self._spinBox.setValue(value)
+        self.percentilesChanged.emit(self.getPercentilesRange())
```

Comment:
As it is, the signal is emitted even if the value is not updated: This should be avoided to avoid infinite signal loops.
Also I think `blockSignals is not needed:

```suggestion
        previousValue = self.getSaturationValue()
        self._slider.setValue(value)
        self._spinBox.setValue(value)
        if value != previousValue:
            self.percentilesChanged.emit(self.getPercentilesRange())
```

previous /current check could also be done on percentile range :shrug:

Reply:
Yes, makes sense!